libparsec/crates/types/src/certif.rs

```diff
@@ -48,10 +48,12 @@ fn dump<T>(obj: &T) -> Vec<u8>
 where
     T: Serialize,
 {
-    let serialized = ::rmp_serde::to_vec_named(obj).unwrap_or_else(|_| unreachable!());
+    let serialized = ::rmp_serde::to_vec_named(obj).expect("object should be serializable");
     let mut e = ZlibEncoder::new(Vec::new(), flate2::Compression::default());
-    e.write_all(&serialized).unwrap_or_else(|_| unreachable!());
-    e.finish().unwrap_or_else(|_| unreachable!())
+    e.write_all(&serialized)
+        .expect("serialized object should be writable to ZlibEncoder");
```

Comment:
An object that impl `Read` will always be writable to `ZlibEncoder` (`object -> read -> zlib -> write -> output`), here the possible error is not about that.

The possible error that can raise here is for the used writer. 

A writer could be anything that impl `Write`, it could be a file, a socket or in this case a `Vec` (thus writing in memory).

Writing to a file or a socket require to pass information to the kernel and with really bad luck can fail (socket is closed, file has no space left to write the data, ...)

In the end, the suggested error message here is not correct.
It should be something like

```suggestion
        .expect("The data is written in memory and should not fail");
```

Reply:
OK, thanks for the explanation